internal/app/go-filecoin/node/node.go

```diff
@@ -234,20 +234,18 @@ func (node *Node) handleNewMiningOutput(ctx context.Context, miningOutCh <-chan
 			return
 		case output, ok := <-miningOutCh:
 			if !ok {
-				return
-			}
-			if output.Err != nil {
-				log.Errorf("stopping mining. error: %s", output.Err.Error())
+				log.Errorf("scheduler stopped. stopping mining.")
 				node.StopMining(context.Background())
-			} else {
-				node.BlockMining.MiningDoneWg.Add(1)
-				go func() {
-					if node.IsMining() {
-						node.BlockMining.AddNewlyMinedBlock(ctx, output)
-					}
-					node.BlockMining.MiningDoneWg.Done()
-				}()
+				return
 			}
+
+			node.BlockMining.MiningDoneWg.Add(1)
+			go func() {
+				if node.IsMining() {
+					node.BlockMining.AddNewlyMinedBlock(ctx, output)
+				}
+				node.BlockMining.MiningDoneWg.Done()
+			}()
```

Comment:
Removing this goroutine later will probably be necessary for us to be able to do rush catch-up mining in a busy loop.